himportscan/src/HImportScan/ImportScanner.hs

```diff
@@ -55,9 +59,15 @@ instance Aeson.ToJSON ScannedImports where
       , ("usesTH", Aeson.toJSON usesTH)
       ]
 
+-- Since it is easier to store in JSON format,
+-- the type `ImportMethod`, isomorphic to `Bool`
+-- is casted during the translation to JSON.
 instance Aeson.ToJSON ModuleImport where
-  toJSON (ModuleImport maybePackageName moduleName) =
-    Aeson.toJSON $ catMaybes [maybePackageName, Just moduleName]
+  toJSON (ModuleImport importMethod maybePackageName moduleName) =
+    Aeson.object
+      [ ("isSourceImported", Aeson.Bool (importMethod == SourceImport))
+      , ("moduleName", Aeson.toJSON $ catMaybes [maybePackageName, Just moduleName])
```

Comment:
Maybe we can make `isSourceImported` present only when the module is source imported. Otherwise the encoding grows for all imports unnecessarily. Booleans are initialized to false in go by default: https://go.dev/tour/basics/12

We could do something similar for the package name, but that can be in a separate PR if you want.